opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -1174,9 +1179,30 @@ namespace Opm
     template<typename TypeTag>
     void
     MultisegmentWell<TypeTag>::
-    addWellContributions(SparseMatrixAdapter& /* jacobian */) const
-    {
-        OPM_THROW(std::runtime_error, "addWellContributions is not supported by multisegment well yet");
+    addWellContributions(SparseMatrixAdapter& jacobian) const
+    {
+        const auto invDuneD = mswellhelpers::invertWithUMFPack<DiagMatWell, BVectorWell>(duneD_, duneDSolver_);
+
+        // We need to change matrix A as follows
+        // A -= C^T D^-1 B
+        // D is diagonal
```

Comment:
This comment does not seem correct as there might be off-diagonal entries for inlet and outlet for each segment.

Reply:
Very true! I started from the StandardWell code and forgot to update the comment.

Reply:
Comment updated.